### PR TITLE
feat: add optional or required decorator on form labels

### PR DESCRIPTION
**Pull request by Jshhhh**
## Problem
Add configured `* or  - optional` to form label

### Manual tests
Add `labelDecorator` property to field config in form schema

### Automated tests
- [x] Unit tests added/updated
- [x] E2E tests added/updated
- [ ] N/A - (provide a reason)
- [ ] deferred - (provide GitHub issue for tracking)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by lyonsbp — DISMISSED** [review]

File: packages/codegen-ui-react/lib/forms/form-renderer-helper/label-decorator.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+const generateLabelDecorator = (labelDecorator: LabelDecorator) => {
+  let styles = [
+    factory.createPropertyAssignment(factory.createIdentifier('whiteSpace'), factory.createStringLiteral('pre')),
+    factory.createPropertyAssignment(factory.createIdentifier('fontStyle'), factory.createStringLiteral('italic')),
+  ];
+  let decoratorText = ' - optional';
+
+  if (labelDecorator === 'required') {
+    styles = [factory.createPropertyAssignment(factory.createIdentifier('color'), factory.createStringLiteral('red'))];
```
Comment (awinberg-aws) [inline packages/codegen-ui-react/lib/forms/form-renderer-helper/label-decorator.ts:81]: can this color be theme-driven?

Comment (Jshhhh) [reply packages/codegen-ui-react/lib/forms/form-renderer-helper/label-decorator.ts:81]: I'm gonna leave it red for now. Gotta think about it more

Comment (Jshhhh) [reply packages/codegen-ui-react/lib/forms/form-renderer-helper/label-decorator.ts:81]: Ill follow up with support for theme

**Review by awinberg-aws — APPROVED** [review]

**Review by awinberg-aws — APPROVED** [review]

File: packages/codegen-ui-react/lib/forms/form-renderer-helper/label-decorator.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      ),
+    ),
+  );
+
+const generateLabelDecorator = (labelDecorator: LabelDecorator) => {
+  let styles = [
+    factory.createPropertyAssignment(factory.createIdentifier('whiteSpace'), factory.createStringLiteral('pre')),
+    factory.createPropertyAssignment(factory.createIdentifier('fontStyle'), factory.createStringLiteral('italic')),
+  ];
+  let decoratorText = ' - optional';
```
[the remark this answers is not in the document]
Comment (Jshhhh) [reply packages/codegen-ui-react/lib/forms/form-renderer-helper/label-decorator.ts:78]: Yeah thats a good point.

**Review by scottyoung — APPROVED** [review]

File: packages/codegen-ui-react/lib/forms/form-renderer-helper/label-decorator.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jshhhh) [reply packages/codegen-ui-react/lib/forms/form-renderer-helper/label-decorator.ts:78]: task created

**Pull request merged**